x-pack/legacy/plugins/siem/public/containers/matrix_histogram/index.ts

```diff
@@ -82,11 +70,8 @@ export const useQuery = <Hit, Aggs, TCache = object>({
         .then(
           result => {
             if (isSubscribed) {
-              const isDataKeyAnArray = Array.isArray(dataKey);
-              const rootDataKey = isDataKeyAnArray ? dataKey[0] : `${dataKey}`;
-              const histogramDataKey = isDataKeyAnArray ? dataKey[1] : `matrixHistogramData`;
-              const source = getOr({}, `data.source.${rootDataKey}`, result);
-              setData(getOr([], histogramDataKey, source));
+              const source = getOr({}, 'data.source.MatrixHistogram', result);
```

Comment:
It won't happen overnight, but we're gradually moving away from using `getOr` in favor of [optional chaining with nullish coalescing](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html), because per the screenshot below, the return type from `getOr` is `any`:

![getOr_returns_any](https://user-images.githubusercontent.com/4459398/74403049-39bfaf80-4de3-11ea-90d9-bbdc7305496c.png)

If we change this statement to:

```
const source = result?.data?.source?.MatrixHistogram ?? {};
```

then per the screenshot below, the type of the `source` variable becomes `GetMatrixHistogramQuery.MatrixHistogram` instead of `any`:

![with-optional-chaining-and-nullish-coalescing](https://user-images.githubusercontent.com/4459398/74403593-febe7b80-4de4-11ea-89d8-2de4049209f2.png)

Consider replacing the use of `getOr` in this file and others with this technique.